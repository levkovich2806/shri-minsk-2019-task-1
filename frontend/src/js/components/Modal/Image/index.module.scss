.image {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;

  &__title {
    font-size: 16px;
    line-height: 22px;
    font-weight: bold;
  }

  &__input {
  }
}
