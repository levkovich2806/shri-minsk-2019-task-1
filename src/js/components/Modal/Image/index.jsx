import React from "react";
import styles from "./index.module.scss";

function Image(props) {
  return (
    <div className={styles.image}>
      <span className={styles.image__title}>Изображение (ссылка): </span>
      <input
        type="text"
        className={styles.image__input}
        onChange={props.onChange}
      />
    </div>
  );
}

export default Image;
