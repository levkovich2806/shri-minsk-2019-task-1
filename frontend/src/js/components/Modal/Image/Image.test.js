import React from 'react';
import { shallow, mount } from 'enzyme';

import Image from './index';

describe('Image', () => {

  it('Should match snapshot Image', () => {
    const wrapper = shallow(<Image url="http://ya.ru/images/logo.png" onChange={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Image check correct input value", () => {

    const wrapper = mount(<Image url="http://ya.ru/images/logo.png" onChange={() => { }} />);

    const input = wrapper.find("input");
    let value = input.get(0).props.value;
    expect(value).toBe("http://ya.ru/images/logo.png");
  });

});
