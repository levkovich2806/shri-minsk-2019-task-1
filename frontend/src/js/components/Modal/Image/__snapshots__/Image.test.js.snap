// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Should match snapshot Image 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    onChange={[Function]}
    url="http://ya.ru/images/logo.png"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="image__title"
        >
          Изображение (ссылка): 
        </span>,
        <input
          className="image__input"
          onChange={[Function]}
          type="text"
          value="http://ya.ru/images/logo.png"
        />,
      ],
      "className": "image",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Изображение (ссылка): ",
          "className": "image__title",
        },
        "ref": null,
        "rendered": "Изображение (ссылка): ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "image__input",
          "onChange": [Function],
          "type": "text",
          "value": "http://ya.ru/images/logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="image__title"
          >
            Изображение (ссылка): 
          </span>,
          <input
            className="image__input"
            onChange={[Function]}
            type="text"
            value="http://ya.ru/images/logo.png"
          />,
        ],
        "className": "image",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Изображение (ссылка): ",
            "className": "image__title",
          },
          "ref": null,
          "rendered": "Изображение (ссылка): ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "image__input",
            "onChange": [Function],
            "type": "text",
            "value": "http://ya.ru/images/logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
